# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [31]:
import pandas as pd
import numpy as np
from scipy import stats
import math

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [24]:
data.shape

(4870, 65)

In [78]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

<hr>
## 1. What test is appropriate for this problem? Does CLT apply?
 - A **two sample t-test is appropriate for this problem**. We want to see if there is a difference between two sample means: the mean callbacks for white-sounding names and the mean callbacks for black-sounding names. 
 - The central limit theorem (CLT) is a statistical theory that states that given a sufficiently large sample size from a population with a finite level of variance, the mean of all samples from the same population will be approximately equal to the mean of the population
  - **In this binomial distribution, since we have a sufficiently large sample size of 4870, the CLT can be applied**.

<hr>
## 2. What are the null and alternate hypotheses?
<u>Hypotheses</u>:

$H$o: Mean callback for white-sounding names = Mean callback for black-sounding names.<br>
$H$a: Mean callback for white-sounding names ≠ Mean callback for black-sounding names.

<hr>
## 3. Compute margin of error, confidence interval, and p-value.
<u>Margin of Error</u>:
- Margin of error = Critical value (z-value) x Standard error
- Standard error = sqrt [variance1/n1 + variance2/n2]
- From the calculations below, we get a **margin of error of 0.015**.

<u>Confidence Interval</u>:
- The range for a **95% confidence interval = +- 0.015**. If the mean difference is above or below 0.015, we can reject the $H$o.
- **Since the mean difference is -0.032 < -0.015, we can reject the $H$o**.

<u>Using T-Statistic</u>:
- **T statistic = -4.1147**
- **P-value = 3.9408**

<u>Decision Rule</u>:
- If p-value < 0.05, we reject  $H$o. From the t test below, we have a small **p-value of 3.9408021031288859e-05 < 0.05, we can reject  H$o$**.

In [28]:
# MEAN CALLBACK OF BLACK AND WHITE, DIFFERENCE IN MEANS

b = data[data.race=='b'].call
w = data[data.race=='w'].call
b_mean = b.mean()
w_mean = w.mean()
diff_mean = b_mean - w_mean
print('Mean callback of black-sounding names:', b_mean)
print('Mean callback of white-sounding names:', w_mean)
print('Difference in means:', diff_mean)


Mean callback of black-sounding names: 0.0644763857126236
Mean callback of white-sounding names: 0.09650924056768417
Difference in means: -0.03203285485506058


In [41]:
b.std()**2

0.06034365571350975

In [40]:
b.var()

0.060343656688928604

In [64]:
# CALCULATE MARGIN OF ERROR = Z-VALUE * STANDARD ERROR

# STANDARD ERROR = sqrt[variance1/n1 + variance2/n2]
v1 = b.var()
v2 = w.var()
v3 = v1+v2/2
n1=len(b)
n2=len(w)
se = math.sqrt(v1/len(b) + v2/len(w))
print('Standard Error:', se)

# MARGIN OF ERROR
moe = 1.96*se
print('Margin of Error:', moe)
print(v3, n1, n2)

Standard Error: 0.007784906826883674
Margin of Error: 0.015258417380692
0.1039581410586834 2435 2435


In [77]:
# CALCULATE T STATISTIC AND P-VALUE USING SCIPY
stats.ttest_ind(b, w)

Ttest_indResult(statistic=-4.1147052908617514, pvalue=3.9408021031288859e-05)

<hr>
## 4. Write a story describing the statistical significance in the context or the original problem.

- From the calculations above, we see that black-sounding names are 3.2% less likely to get a callback than white-sounding names, taken from the mean difference (diff_mean) above. 

- Using the two sample t statistic, we get a significantly small p-value of 0.0000394,  we can conclude that with more than 95% confidence, black-sounding names have a lower chance of getting a callback than white-sounding names.

<hr>
## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

- From my analysis, it is apparent that race/name DOES have an effect on callback success. However, I cannot say that it is the MOST important factor because I have not analyzed other factors that can affect callback success.
- To figure out if race/name is the most important factor, further analysis should be performed on these other factors like education and years of experience.